scripts/zones/Arrapago_Reef/npcs/_ji7.lua

```diff
@@ -0,0 +1,45 @@
+-----------------------------------
+-- Area: Arrapago Reef
+-- Door: Tidal Gate (Medusa Arena)
+-- !pos -397 -15 420
+-----------------------------------
+local ID = require("scripts/zones/Arrapago_Reef/IDs")
+require("scripts/globals/npc_util")
+-----------------------------------
+local entity = {}
+
+entity.onTrade = function(player, npc, trade)
+    local item = xi.items.LAMIAN_CLAW_KEY
+
+    if
+        player:getXPos() >= -405 and
+        npcUtil.tradeHasExactly(trade, { item, item + 1, item + 2 })
+    then
+        player:confirmTrade()
+        npc:openDoor(15)
+        player:messageSpecial(zones[npc:getZoneID()].text.KEYS_SHATTER, item, item + 1, item + 2)
+    end
+end
+
+entity.onTrigger = function(player, npc)
+    if
+        player:getXPos() <= -407 and
+        npc:getAnimation() == xi.anim.CLOSE_DOOR
+    then
+        player:startEvent(229)
+    end
+end
+
+entity.onEventUpdate = function(player, csid, option)
+end
+
+entity.onEventFinish = function(player, csid, option)
+    if
+        csid == 229 and
+        option == 1
+    then
+        player:setPos(-322.6, -10.2, 319.5, 214, 54) -- Teleport destination needs confirmation
```

Comment:
Do you need a capture of this?

Reply:
yess <3 please!

Reply:
hope multiple people have keys or are ready to go as soon as its opened.....

Reply:
x -90.910 y -2.645 z -612.908 Rot 192 zone 54

Reply:
MUCH LOVE